### DISCUSSION TITLE
Is it possible to parse args and options of commands when running subcommands?

### DISCUSSION THREAD

**Discussion by ottobolyos** [jmcdo29/nest-commander#888] [Q&A]
I want to have som options that are accepted by a command and I should be able to use them in subcommands.

For example:

- [command] `plc`:
  - [option] `-i, --ip <string>`;
- [subcommand] `get`.

Now, when I run `app plc get -i 1.1.1.1`, the `options` array in `GetCommand#run` is not in there. Moreover, if I add `-i, --ip <string>` option to `get` subcommand, Nest Commander associates the option with `plc` command only, regardless of the command-option-subcommand order (i.e. it does not matter if it is `${command} ${option} ${subcommand}` or `${command} ${subcommand} ${option}`).

Use case: when some options are used by all subcommands and we want to make it KISS-compliant.

**Comment by jmcdo29** [comment]
I believe you're looking for something like [this issue](https://github.com/jmcdo29/nest-commander/issues/663). Generally, I don't recall if Commander allows for sub commands to inherit the options from the parent commands.

**Reply by tukusejssirs** [reply]
#663 (and tj/commander.js#1824) is a related issue, but in this use case I actually want to _inherit_ all ancestor (sub-) commands options by a subcommand.

I believe this is achievable in Nest Commander, however, I think it might be better to be supported by `commander` itself.

Of course, this feature should be behind a global option in `CommandFactory#run` and related methods in order to don’t break the previous behaviour.

Now, in `commander`, I’ve found tj/commander.js#1753, howerver, based on [this comment](https://github.com/tj/commander.js/issues/1753#issuecomment-1186102211), it seems like they are not interested in this feature ATM.

Are you interested in this feature to be implemented in Nest Commander? Is there a workaround to inherit all ancestor commands options in a subcommand? Or do I need to persuade `commander` devs to include such a feature? :thinking:

**Reply by tukusejssirs** [reply]
Actually, it is somewhat possible to do in `commander`, see [`global-options-nested.js`](https://github.com/tj/commander.js/blob/33195f189b1ffa568c232503fb6ac0cf5548eb74/examples/global-options-nested.js) and [`global-options-nested.js`](https://github.com/tj/commander.js/blob/33195f189b1ffa568c232503fb6ac0cf5548eb74/examples/global-options-added.js) examples, neither is perfect, but I think we could make it work.

**Reply by jmcdo29** [reply]
Ah, looking at those examples it should be easy enough to add them in via those examples. I'll probably need to think it through a bit to find the most maintainable API
